spanner/google/cloud/spanner/types.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Types exported from this package."""
+
+from google.cloud.proto.spanner.v1 import type_pb2
+
+
+# Scalar paramter types
+STRING_PARAM_TYPE = type_pb2.Type(code=type_pb2.STRING)
```

Comment:
Ideally, I would like to use the identical pattern we are using elsewhere, which has a few nice usability wins:

  1. It uses the identical type names to the `_pb2` modules.
  2. It rewrites `__module__` which helps users when looking at docs.
  3. It grabs all types from the modules on the fly, so if Spanner introduces new types, it does not need manual updating.

[Here](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/vision/google/cloud/vision_v1/types.py#L23-L35) is what I did in Vision. Would you have any objection to doing that here?